src/mono/mono/mini/monovm.c

```diff
@@ -276,3 +277,89 @@ monovm_shutdown (int *latchedExitCode)
 
 	return 0;
 }
+
+static int
+monovm_create_delegate_impl (const char* assemblyName, const char* typeName, const char *methodName, void **delegate);
+	
+
+int
+monovm_create_delegate (const char *assemblyName, const char *typeName, const char *methodName, void **delegate)
+{
+	int result;
+	/* monovm_create_delegate may be called instead of monovm_execute_assembly.  Initialize the
+	 * runtime if it isn't already. */
+	if (!mono_get_root_domain())
+		mini_init (assemblyName, "v4.0.30319");
+	MONO_ENTER_GC_UNSAFE;
+	result = monovm_create_delegate_impl (assemblyName, typeName, methodName, delegate);
+	MONO_EXIT_GC_UNSAFE;
+	return result;
+}
+
+int
+monovm_create_delegate_impl (const char* assemblyName, const char* typeName, const char *methodName, void **delegate)
+{
+	// Load an assembly and a type and a method from the type, then return a pointer to a native
+	// callable version of the method.  See CorHost2::CreateDelegate
+	//
+	// We have to do this in general, but the CoreCLR hostpolicy only calls this with a handful
+	// of methods from the [CoreLib]Internal.Runtime.InteropServices.ComponentActivator
+	// class. See hostpolicy.cpp get_delegate()
+
+	ERROR_DECL (error);
+
+	const int failure = 0x80004005; /* E_FAIL */
+
+	if (!delegate)
```

Comment:
@lambdageek To avoid user mistakes earlier, you should also be able to validate `methodName` is set to `UNMANAGEDCALLERSONLY_METHOD`.

https://github.com/dotnet/runtime/blob/025c84dd0096214b00a12d6058f4554163fbc721/src/native/corehost/coreclr_delegates.h#L21-L32

Reply:
@AaronRobinsonMSFT the method passed to `load_assembly_and_get_function_pointers` doesn't have to be `UnmanagedCallersOnly` on mono - that's not what I'm checking here.  `load_assembly_and_get_function_pointer` is the same one from ComponentActivator that CoreCLR uses.  You can pass a normal managed method and a delegate type and it will work.

This is just checking that the destination outarg isn't null before we write into it.

Reply:
I realize that isn't what is being checked here, but it is earlier - that was my point relative to the method below being checked for the `UnmanagedCallersOnly` attribute. However, I mistakenly stated the method name when I really mean the delegate name.

Reply:
I misinterpreted this code path entirely.